students/smitco/lesson09/mailroomOO.py

```diff
@@ -0,0 +1,147 @@
+# lesson 09 mailroom object oriented
+# !/usr/bin/env python3
+
+
+import os
+import datetime
+
+donors = {"John Travolta": [5000, 7500],
+          "Jane Fonda": [10000, 8000, 6500],
+          "Judy Blume": [3000, 3000],
+          "Joey Tribbiani": [9000],
+          "Jenny Gump": [10300, 13750, 12500]}
+
+class Donor(object):
+    def __init__(self, name, *args, **kwargs):
+       self.name = name
+    
+    def donor_name(self, name):
+        if self.name.lower() == "list":
+            print(list(donors.keys()))
+        elif self.name.lower() == "exit":
+            print("\nExiting.\n")
+        return self.name
+    
+    def donor_amount(self, name, amount):
+        self.amount = amount
+        try:
+            if self.amount.lower() == "exit":
+                print("\nExiting.")
+                self.amount = "exit"
+            elif int(self.amount) >= 10**6:
+                print("\nThe amount entered is too large.")
+                self.amount = "large"
+            else:
+                self.amount = int(self.amount)
+        except ValueError:
+            print("\nInvalid entry.")
+            self.amount = "invalid"
+        return self.amount
+
+    def donor_append(self, name, amount):
+        if self.name in donors:
+            donors[self.name].append(self.amount)
+        else:
+            donors[self.name] = [self.amount]
+
+    def donor_email(self, name, amount):
+        print("\nThank you, {}, for your generous donation of ${} "
+              "to the Brave Heart Foundation.".format(self.name, self.amount))
+
+class All_Donors(Donor):
```

Comment:
class naming convention is `UpperCamelCase`

Reply:
AllDonors should _not_ inherit from Donor; inheritance relationship is more like Student (child class) *is* Person (parent class).

The relationship between `Donor` and `AllDonors` is one to many; meaning `AllDonors` should have a reference to collection of donor objects.

Since you're still using global dictionary to hold data, you're not really using the classes to hold any data asked in the assignment.